connector/servicegraphconnector/factory.go

```diff
@@ -6,13 +6,80 @@
 package servicegraphconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector"
 
 import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/featuregate"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector/internal/metadata"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/servicegraphprocessor"
 )
 
+const (
+	// The value of "type" key in configuration.
+	typeStr = "servicegraph"
+	// The stability level of the processor.
+	connectorStability                    = component.StabilityLevelDevelopment
+	virtualNodeFeatureGateID              = "processor.servicegraph.virtualNode"
+	legacyLatencyMetricNamesFeatureGateID = "processor.servicegraph.legacyLatencyMetricNames"
+	legacyLatencyUnitMs                   = "processor.servicegraph.legacyLatencyUnitMs"
```

Comment:
my only thought around keeping these the same is to prevent having to migrate feature gates that I think we want to deprecate... maybe its worth deprecating the feature gates before removing the component? 

It's a bit tricky, because the connector currently uses these gates as they are, so if a user is configuring the connector as of v0.92.0 with `processor.servicegraph.legacyLatencyMetricNames`, that same configuration wouldn't work with a version where the gate is renamed to `connector.servicegraph.legacyLatencyMetricNames`

Reply:
Hmm, I see. I'm not very familiar with the process of deprecating feature gates. I think you're right and we should do that at a different PR.

Reply:
If the connector was already using the `processor.*` flags, then it should keep using them and renaming those feature flags is not really related to removing the processor. Not sure if renaming the feature gates makes sense, as features gates are temporary in nature. By the way, I'm not seeing these feature gates mentioned in either the processor's or the connector's README. 🤔

Reply:
will keep the feature gate as is. hopefully we can remove them sooner than later and avoid future confusion